File: Lista de Exercicio Paradigmas/Lista Java/Ex12.java
class Produto {
    private String nome;
    private double preco;

    public Produto(String nome, double preco) {
        this.nome = nome;
        this.preco = preco;
    }

    public Produto add(Produto outro) {
        if (outro instanceof Produto) {
            String novoNome = this.nome + " + " + outro.nome;
            double novoPreco = this.preco + outro.preco;
            return new Produto(novoNome, novoPreco);
        }
        return null;
    }

    @Override
    public String toString() {
        return "Produto(nome='" + nome + "', preco=" + String.format("%.2f", preco) + ")";
    }

    public static void main(String[] args) {
        Produto produto1 = new Produto("Produto A", 20.50);
        Produto produto2 = new Produto("Produto B", 15.75);

        Produto produtoSoma = produto1.add(produto2);
        System.out.println(produtoSoma);
    }
}

File: Lista de Exercicio Paradigmas/Lista Python/Ex12.py
class Produto:
    def __init__(self, nome, preco):
        self.nome = nome
        self.preco = preco

    def __add__(self, outro):
        if isinstance(outro, Produto):
            novo_nome = f"{self.nome} + {outro.nome}"
            novo_preco = self.preco + outro.preco
            return Produto(novo_nome, novo_preco)
        return NotImplemented

    def __repr__(self):
        return f"Produto(nome='{self.nome}', preco={self.preco:.2f})"

# Exemplo de uso
produto1 = Produto("Produto A", 20.50)
produto2 = Produto("Produto B", 15.75)

produto_soma = produto1 + produto2
print(produto_soma)